### PR TITLE
require `--yes` flag to promote preview deployment

### PR DESCRIPTION
**Pull request by EndangeredMassa**
In order to promote a preview deployment, we want to make sure the user knows that this is not typical. Now the `--yes` flag is required to make this work.

### CODE REVIEW

**Review by cb1kenobi — DISMISSED** [review]
LGTM!

**Review by cb1kenobi — DISMISSED** [review]

**Review by styfle — CHANGES_REQUESTED** [review]
Let's use `--yes` for consistency with other commands

**Comment by cb1kenobi** [comment]
> Let's use `--yes` for consistency with other commands

Personally, as much as I love consistency, I think `--force` makes more sense than `--yes` in this case.

File: packages/cli/src/commands/promote/request-promote.ts [hunk 1]
```diff
@@ -29,6 +31,13 @@ export default async function requestPromote({
     output: client.output,
   });
 
+  if (deployment.target !== 'production' && !force) {
+    output.error(
+      'Cannot promote deployment that does not target "production". If you are sure you want to do this, add `--force`.'
+    );
```
Comment (styfle) [inline packages/cli/src/commands/promote/request-promote.ts]: It seems contradictory to say “cannot” and then say that you can with a flag. 

I think we can improve this message and include reasons why you wouldn’t want to do this. 

Something like:

> Warning: This deployment does not target production, therefore promotion will not apply production environment variables. Are you sure you want to continue? Y/n

Comment (EndangeredMassa) [reply packages/cli/src/commands/promote/request-promote.ts]: I like this better. In this framing, it does make more sense to use `--yes`.

Comment (EndangeredMassa) [reply packages/cli/src/commands/promote/request-promote.ts]: Switched! https://github.com/vercel/vercel/pull/10135/commits/de9914719cc5fca88e6977447550862eafa16c74

File: packages/cli/src/commands/promote/request-promote.ts [hunk 2]
```diff
@@ -29,6 +32,18 @@ export default async function requestPromote({
...
 
+  if (deployment.target !== 'production' && !yes) {
+    const question =
+      'This deployment does not target production, therefore promotion will not apply production environment variables. Are you sure you want to continue?';
+    const answer = await confirm(client, question, true);
+    if (!answer) {
+      output.error(
+        'User declined to promote a deployment that did not target production.'
+      );
+      return 1;
```
Comment (styfle) [inline packages/cli/src/commands/promote/request-promote.ts]:
```suggestion
    const answer = await confirm(client, question, false);
    if (!answer) {
      output.error('Canceled');
      return 0;
```

Looks like the pattern here is to default to `false` and then false will print Canceled with 0 exit code.

See other commands here:

https://github.com/vercel/vercel/blob/36db62a491aa3155ebb286f0e533bd8edfc93670/packages/cli/src/commands/env/rm.ts#L104-L116

https://github.com/vercel/vercel/blob/36db62a491aa3155ebb286f0e533bd8edfc93670/packages/cli/src/commands/env/pull.ts#L88-L97

File: .changeset/green-badgers-reflect.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"vercel": patch
+---
+
+require force to promote preview deployment
```
Comment (styfle) [inline .changeset/green-badgers-reflect.md]:
```suggestion
require `--yes` to promote preview deployment
```

**Review by styfle — APPROVED** [review]
Thanks!

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**